### PR TITLE
[mysticeti] add SuiBlockValidator

### PR DESCRIPTION
**Pull request by longbowlu**
## Description 

This PR imports mysticeti and implements `BlockValidator` for `SuiBlockValidator`. The former is the counterpart of `TransactionValidator` in narwhal, used to verify the block contents. The major verification logic is extracted into a shared function `validate_transactions`.

Also updated `futures` and `zeroize`'s config to align with mysticeti's. 

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by andll — COMMENTED** [review]
Overall looks good, but I wonder if it would be easier, if instead of introducing new type we just `impl BlockValidator for SuiTxValidator`.
In that case `validate_transactions` could be just instance method on the `SuiTxValidator`, so we don't need to pass all the parameters like epoch_store in this function

**Comment by longbowlu** [comment]
> Overall looks good, but I wonder if it would be easier, if instead of introducing new type we just `impl BlockValidator for SuiTxValidator`. In that case `validate_transactions` could be just instance method on the `SuiTxValidator`, so we don't need to pass all the parameters like epoch_store in this function

there was a reason that I chose to add a new type but I can't recall now... let me think for a bit and if nothing comes up will use the same type.

**Review by mwtian — APPROVED** [review]

**Review by andll — APPROVED** [review]

File: crates/sui-core/src/consensus_validator.rs [hunk 1]
```diff
@@ -141,6 +115,55 @@ impl TransactionValidator for SuiTxValidator {
...
+        self.validate_transactions(txs).await
+    }
+}
+
+#[async_trait]
+impl BlockValidator for SuiTxValidator {
+    type Error = eyre::Report;
+
+    async fn verify(&self, b: &StatementBlock) -> Result<(), Self::Error> {
+        let txs = b
```
Comment (akichidis) [inline crates/sui-core/src/consensus_validator.rs:158]: maybe we can similarly track the performance here via `let _scope = monitored_scope("ValidateBlock");`

File: crates/sui-core/src/generate_format.rs [hunk 2]
```diff
@@ -136,6 +136,9 @@ fn get_registry() -> Result<Registry> {
     let ccd = CheckpointContentsDigest::random();
     tracer.trace_value(&mut samples, &ccd)?;
 
+    let struct_tag = StructTag::from_str("0x2::coin::Coin<0x2::sui::SUI>").unwrap();
+    tracer.trace_value(&mut samples, &struct_tag)?;
```
Comment (akichidis) [inline crates/sui-core/src/generate_format.rs:140]: Why is this needed?

File: crates/sui-core/src/consensus_validator.rs [hunk 3]
```diff
@@ -141,6 +115,55 @@ impl TransactionValidator for SuiTxValidator {
...
+            .iter()
+            .map(|tx| tx_from_bytes(tx).map(|tx| tx.kind))
+            .collect::<Result<Vec<_>, _>>()?;
+
+        self.validate_transactions(txs).await
+    }
+}
+
+#[async_trait]
+impl BlockValidator for SuiTxValidator {
```
Comment (akichidis) [inline crates/sui-core/src/consensus_validator.rs:154]: Nice, this aligns well with the expectations of https://github.com/MystenLabs/sui/pull/14559

**Review by akichidis — APPROVED** [review]

File: crates/sui-core/src/generate_format.rs (hunk 2 above)
Comment (longbowlu) [reply crates/sui-core/src/generate_format.rs:140]: I don't have the intrinsic reason but upgrading serde from 1.0.152 to any later versions will panic if this is not added.

Comment (akichidis) [reply crates/sui-core/src/generate_format.rs:140]: ah I see, got it
